src/librustdoc/html/render/mod.rs

```diff
@@ -798,17 +798,67 @@ function handleThemeButtonsBlur(e) {{
     var active = document.activeElement;
     var related = e.relatedTarget;
 
-    if (active.id !== "themePicker" &&
+    if (active.id !== "theme-picker" &&
         (!active.parentNode || active.parentNode.id !== "theme-choices") &&
         (!related ||
-         (related.id !== "themePicker" &&
+         (related.id !== "theme-picker" &&
           (!related.parentNode || related.parentNode.id !== "theme-choices")))) {{
         hideThemeButtonState();
     }}
 }}
 
+function handleThemeKeyDown(e) {{
+    if (e.altKey || e.ctrlKey || e.metaKey || e.shiftKey) {{ return; }}
```

Comment:
Two things:

1. No inlined code please.
2. This seems to be a "non generic" code. I think it should be put in a JS file directly.

About `2.`: if it makes sense we can maybe create the equivalent JS file and then update the generic part of it.

Reply:
Okay, I've fixed the style thing, and pulled the new code out into main.js, with the rest of the non-generic code.

Reply:
Fine by me!